src/code/z_kankyo.c

```diff
@@ -1,94 +1,2499 @@
+    gSPDisplayList(POLY_XLU_DISP++, gFieldSandstormDL);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_kankyo.c", 4068);
+
+    D_8015FDB0 += (s32)sp98;
+}
+
+void Kankyo_AdjustLights(GlobalContext* globalCtx, f32 arg1, f32 arg2, f32 arg3, f32 arg4) {
+    f32 temp;
+    s32 i;
+
+    if (globalCtx->roomCtx.curRoom.unk_03 != 5 && func_800C0CB8(globalCtx)) {
+        arg1 = CLAMP_MIN(arg1, 0.0f);
+        arg1 = CLAMP_MAX(arg1, 1.0f);
+
+        temp = arg1 - arg3;
+        if (arg1 < arg3) {
+            temp = 0.0f;
+        }
+
+        globalCtx->envCtx.adjFogNear = (arg2 - globalCtx->envCtx.lightSettings.fogNear) * temp;
+
+        if (arg1 == 0.0f) {
+            for (i = 0; i < 3; i++) {
+                globalCtx->envCtx.adjFogColor[i] = 0;
+            }
+        } else {
+            temp = arg1 * 5.0f;
+            temp = CLAMP_MAX(temp, 1.0f);
+
+            for (i = 0; i < 3; i++) {
+                globalCtx->envCtx.adjFogColor[i] = -(s16)(globalCtx->envCtx.lightSettings.fogColor[i] * temp);
+            }
+        }
+
+        if (arg4 <= 0.0f) {
+            return;
+        }
+
+        arg1 *= arg4;
+
+        for (i = 0; i < 3; i++) {
+            globalCtx->envCtx.adjAmbientColor[i] = -(s16)(globalCtx->envCtx.lightSettings.ambientColor[i] * arg1);
+            globalCtx->envCtx.adjLight1Color[i] = -(s16)(globalCtx->envCtx.lightSettings.light1Color[i] * arg1);
+        }
+    }
+}
+
+s32 Kankyo_GetBgsDayCount(void) {
+    return gSaveContext.bgsDayCount;
+}
+
+void Kankyo_ClearBgsDayCount(void) {
+    gSaveContext.bgsDayCount = 0;
+}
+
+s32 Kankyo_GetTotalDays(void) {
+    return gSaveContext.totalDays;
 }
 
 void func_800775F0(u16 arg0) {
     gSaveContext.unk_140E = arg0;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/func_80077600.s")
+s32 func_80077600(void) {
+    s32 ret = false;
+
+    if (gSaveContext.unk_140E == 0xFFFF) {
+        ret = true;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/func_80077624.s")
+    return ret;
+}
+
+void func_80077624(GlobalContext* globalCtx) {
+    if (globalCtx->soundCtx.nightSeqIndex == 19) {
+        func_800F6FB4(5);
+    } else {
+        func_800F6FB4(globalCtx->soundCtx.nightSeqIndex);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/func_80077684.s")
+    func_800F6D58(14, 1, 1);
+    func_800F6D58(15, 1, 1);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/func_800776E4.s")
+void func_80077684(GlobalContext* globalCtx) {
+    func_800F6D58(14, 1, 0);
+    func_800F6D58(15, 1, 0);
+
+    if (func_800FA0B4(0) == 1) {
+        gSaveContext.seqIndex = 0x80;
+        func_800758AC(globalCtx);
+    }
+}
+
+/**
+ * After a warp song has been played, set the next entrance accordingly and begin the fade out.
+ * Additionally, set the flag for having watched the intro cutscene for the destination area.
+ * This makes it so the cutscene will be skipped if you have already been to the area via warp song.
```

Comment:
I'm not sure I understand this right. This sounds like "works as expected", did this mean
```suggestion
 * After a warp song has been played, set the next entrance accordingly and begin the fade out.
 * Additionally, set the flag for having watched the intro cutscene for the destination area.
 * This makes it so the cutscene will be skipped if you visit an area for the first time via warp song.
```

Reply:
My choice of words is because no matter what, the entrance cutscene will not play if you enter via a warp song. Setting the flag is so that the next time you enter the area from the "normal entrance", the cutscene should not play because youve *already been to* the area (via warp song)

Reply:
actually i might be mistaken. my understanding was that the warp songs have unique entrances, and the cutscene is only looking for the entrance to the area on foot. But im seeing an entries in `sEntranceCutsceneTable` for warp song entrances lol

so, at first they wanted to have entrance cutscenes from warp songs and then "removed" them by setting the flags in this manner? idk

Reply:
or wait, is the entrance cutscene for the warp song entrance is the warp in cutscene? i know those are scripted cutscenes so i guess that makes sense lol

in that case setting the flags here is pointless cause z_scene has these same flags in its table? 
im about to remove my comment from the function cause this is weird lol

Reply:
as said in discord, just removing this comment. i have no idea whats going on anymore lol
will revisit some other time